File: 1-TwoSum/1-TwoSum.cs
// Last updated: 6/12/2025, 8:28:30 PM
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int x=0;
        int[] dummy=new int[2];
        for (int i=0;i<nums.Length;i++)
        {
            x=nums[i];
            
            for(int j=i+1;j<nums.Length;j++)
            {
                if(nums[j]==target-x)
                {
                    dummy[0]=i;
                    dummy[1]=j;
                    break;
                }
            }
        }
        return dummy;
        
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 6/12/2025, 8:28:29 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int diff{};
        vector<int> ret{};
        unordered_map<int,int> dict;
        for(int i=0;i<nums.size();i++)
        {   diff=target-nums[i];
            if(dict.find(diff)!=dict.end())//if not found maps return the last index
            {
                ret.push_back(i);
                ret.push_back(dict[diff]);
                return ret;
            }
            dict[nums[i]]=i;
        }
        return ret;
    }
};